shared-bindings/adcbuffer/BufferedInput.c

```diff
@@ -0,0 +1,180 @@
+
+    // Create local object
+    adcbuffer_bufferedinput_obj_t *self = m_new_obj(adcbuffer_bufferedinput_obj_t);
+    self->base.type = &adcbuffer_bufferedinput_type;
+
+    // Call local intereface in ports/common-hal/adcbuffer
+    common_hal_adcbuffer_bufferedinput_construct(self,
+        pin,
+        ((uint8_t *)bufinfo.buf),
+        bufinfo.len,
+        bytes_per_sample,
+        signed_samples,
+        sample_rate
+        );
+
+    return MP_OBJ_FROM_PTR(self);
+}
+
+//|     def deinit(self) -> None:
+//|         """Turn off the BufferedInput and release the pin for other use."""
+//|         ...
+//|
+STATIC mp_obj_t adcbuffer_bufferedinput_deinit(mp_obj_t self_in) {
+    adcbuffer_bufferedinput_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    common_hal_adcbuffer_bufferedinput_deinit(self);
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_1(adcbuffer_bufferedinput_deinit_obj, adcbuffer_bufferedinput_deinit);
+
+STATIC void check_for_deinit(adcbuffer_bufferedinput_obj_t *self) {
+    if (common_hal_adcbuffer_bufferedinput_deinited(self)) {
+        raise_deinited_error();
+    }
+}
+
+//|  Provided by context manager helper.
+//|
+//|     def __exit__(self) -> None:
+//|         """Automatically deinitializes the hardware when exiting a context. See
+//|         :ref:`lifetime-and-contextmanagers` for more info."""
+//|         ...
+//|
+STATIC mp_obj_t adcbuffer_bufferedinput___exit__(size_t n_args, const mp_obj_t *args) {
+    (void)n_args;
+    common_hal_adcbuffer_bufferedinput_deinit(args[0]);
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(adcbuffer_bufferedinput___exit___obj, 4, 4, adcbuffer_bufferedinput___exit__);
+
+//|     value: None
+//|         """Fills the supplied buffer with ADC values using DMA transfer.
+//|     If the buffer is 8-bit, then values are 8-bit shifted and error bit is off.
+//|     If buffer is 16-bit, then values are not shifted and error bit is present.
+//|     Number of transfers is always the number of samples which is the array
+//|     byte length divided by the bytes_per_sample."""
+//|
+STATIC mp_obj_t adcbuffer_bufferedinput_obj_readmultiple(mp_obj_t self_in) {
+    adcbuffer_bufferedinput_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    check_for_deinit(self);
+    common_hal_adcbuffer_bufferedinput_readmultiple(self);
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_1(adcbuffer_bufferedinput_readmultiple_obj, adcbuffer_bufferedinput_obj_readmultiple);
+
+STATIC const mp_rom_map_elem_t adcbuffer_bufferedinput_locals_dict_table[] = {
+    { MP_ROM_QSTR(MP_QSTR_deinit),             MP_ROM_PTR(&adcbuffer_bufferedinput_deinit_obj) },
+    { MP_ROM_QSTR(MP_QSTR___enter__),          MP_ROM_PTR(&default___enter___obj) },
+    { MP_ROM_QSTR(MP_QSTR___exit__),           MP_ROM_PTR(&adcbuffer_bufferedinput___exit___obj) },
+    { MP_ROM_QSTR(MP_QSTR_readmultiple),       MP_ROM_PTR(&adcbuffer_bufferedinput_readmultiple_obj)},
```

Comment:
Rather than call this `readmultiple()`, I'd suggest just calling it `read()`. There are plenty of examples of `read()` reading multiple values, such as in `UART`, etc.

Reply:
Changed as suggested.